core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -97,9 +97,9 @@ private[spark] class Executor(
   // Set the classloader for serializer
   env.serializer.setDefaultClassLoader(replClassLoader)
 
-  // Max RPC message size. If task result is bigger than this, we use the block manager
+  // Max size of direct result. If task result is bigger than this, we use the block manager
   // to send the result back.
-  private val maxRpcMessageSize = RpcUtils.maxMessageSizeBytes(conf)
+  private val maxDirectResultSize = Math.min(1L << 20, RpcUtils.maxMessageSizeBytes(conf))
```

Comment:
we should have another config for this instead. Also 1MB is very small, maybe 16 or 32MB?